File: go/graphql/field/hello.go
package graphqlexamples

import (
	"fmt"

	"github.com/graphql-go/graphql"
)

// HelloField define a hello query field
var HelloField *graphql.Field

// ByeField define a bye query field
var ByeField *graphql.Field

func init() {
	HelloField = &graphql.Field{
		Type: graphql.String,
		Args: graphql.FieldConfigArgument{
			"message": &graphql.ArgumentConfig{
				Type: graphql.String,
			},
		},
		Resolve: func(p graphql.ResolveParams) (interface{}, error) {
			message, ok := p.Args["message"].(string)
			if !ok {
				return "Can not read 'message'!", nil
			}
			return fmt.Sprintf("Hello %s!", message), nil
		},
	}
	ByeField = &graphql.Field{
		Type: graphql.String,
	}
}

File: php/graphql/field/Hello.php
<?php
/**
 * User: WindomZ
 * Date: 17-7-20
 */

namespace GraphQLExamples\Field;

use GraphQL\Type\Definition\Type;

class Hello
{
    public static function getHelloField()
    {
        return [
            'type' => Type::string(),
            'args' => [
                'message' => ['type' => Type::string()],
            ],
            'resolve' => function ($root, $args) {
                return 'Hello ' . $args['message'] . '!';
            }
        ];
    }

    public static function getByeField()
    {
        return [
            'type' => Type::string(),
//        'resolve' => function ($root, $args) {
//            return $root['bye'];
//        }
        ];
    }
}

